packages/docusaurus-theme-search-algolia/src/index.js

```diff
@@ -67,6 +67,13 @@ function theme(context) {
               href: normalizeUrl([baseUrl, OPEN_SEARCH_FILENAME]),
             },
           },
+          {
+            tagName: 'meta',
+            attributes: {
+              name: 'robots',
+              content: 'noindex, follow',
+            },
+          },
```

Comment:
Unfortunately this adds noindex nofollow to ALL the docusaurus site, not just search.

Reply:
How do we do it just to search? That's what we want. 

And I don't understand why this isn't a good idea to do for all search results pages using Docusaurus, as that's long been the recommendation of all major search engines.

Reply:
Because this change does not prevent indexing the search page, it prevents indexing ALL the documentation website (ie the website would disappear from Google, very bad!).
It has never been advised by search engines to not index anything at all.

How to apply this only to a single page? Not through the config, but on the page itself you can use this declarative component:

https://v2.docusaurus.io/docs/docusaurus-core#head

Read the doc of React helmet to understand how it works.

Reply:
@slorber I wasn't the one doing the pull, my colleague @jcomack did that, but I think he thought (as I do, reading his code) that he's just changing the theme of the search page. If it doesn't, then by all means do the change that makes it just noindex the search results page :)